x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyRecord.java

```diff
@@ -290,11 +302,18 @@ XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws I
             builder.field(OVER_FIELD_VALUE.getPreferredName(), overFieldValue);
         }
         if (causes != null) {
-            builder.field(CAUSES.getPreferredName(), causes);
+            builder.startArray(CAUSES.getPreferredName());
+            for (AnomalyCause cause : causes) {
+                builder.value(cause, params);
```

Comment:
This line could be changed to `cause.toXContent(builder, params);` to avoid the need to change method visibility in the X-Content builder.